src/routing/approved.rs

```diff
@@ -2031,73 +2036,62 @@ impl Approved {
         Ok(commands)
     }
 
-    pub fn send_user_message(
-        &mut self,
-        src: SrcLocation,
-        dst: DstLocation,
-        content: Bytes,
-    ) -> Result<Vec<Command>> {
-        if !src.contains(&self.node.name()) {
+    pub fn send_user_message(&mut self, itry: Itinerary, content: Bytes) -> Result<Vec<Command>> {
+        let are_we_src =
+            matches!(itry.src, SrcLocation::Node(_)) && itry.src.name() == self.node.name();
+        if !are_we_src {
             error!(
-                "Not sending user message {:?} -> {:?}: not part of the source location",
-                src, dst
+                "Not sending user message {:?} -> {:?}: we are not the source location",
+                itry.src, itry.dst
             );
             return Err(Error::InvalidSrcLocation);
         }
-
-        if matches!(dst, DstLocation::Direct) {
+        if matches!(itry.src, SrcLocation::EndUser(_)) {
```

Comment:
this check won't be hit as the above check already filter this out?

Reply:
Yep. No change required with the fix resulting from the above comment though 👍